app/code/Magento/CatalogMessageBroker/etc/di.xml

```diff
@@ -50,4 +50,29 @@
             <argument name="batchSize" xsi:type="number">500</argument>
         </arguments>
     </type>
+
+    <type name="Magento\CatalogMessageBroker\Model\StorefrontConnector\Connector">
+        <arguments>
+            <argument name="connectionType" xsi:type="string">in-memory</argument>
+        </arguments>
+    </type>
+
+    <type name="Magento\CatalogMessageBroker\Model\StorefrontConnector\ConnectionPool">
+        <arguments>
+            <argument name="connectionMap" xsi:type="array">
+                <item name="in-process" xsi:type="string">Magento\CatalogStorefrontApi\Api\Catalog</item>
```

Comment:
"in-process" means someting inside "the same" process, here we do call "out" of our process - over network. Suggest to rename to "network" or "grpc"

```suggestion
                <item name="network" xsi:type="string">Magento\CatalogStorefrontApi\Api\Catalog</item>
```